edb/server/protocol/ai_ext.py

```diff
@@ -1174,25 +1214,69 @@ async def aconnect_sse(
 
 
 async def _start_openai_like_chat(
+    *,
     protocol: protocol.HttpProtocol,
     request: protocol.HttpRequest,
     response: protocol.HttpResponse,
     client: http.HttpClient,
     model_name: str,
     messages: list[dict[str, Any]],
     stream: bool,
+    temperature: Optional[float],
+    top_p: Optional[float],
+    max_tokens: Optional[int],
+    seed: Optional[int],
+    safe_prompt: Optional[bool],
+    logit_bias: Optional[dict[int, int]],
+    logprobs: Optional[bool],
+    user: Optional[str],
+    tools: Optional[list[dict[str, Any]]],
 ) -> None:
+    isOpenAI = "openai" in str(getattr(client, "base_url", ""))
+    params: dict[str, Any] = {
+        "model": model_name,
+        "messages": messages,
+    }
+    if temperature is not None:
+        params["temperature"] = temperature
+    if top_p is not None:
+        params["top_p"] = top_p
+    if tools is not None:
+        params["tools"] = tools
+    if isOpenAI and logit_bias is not None:
+        params["logit_bias"] = logit_bias
+    if isOpenAI and logprobs is not None:
+        params["logprobs"] = logprobs
+    if isOpenAI and user is not None:
+        params["user"] = user
+    if not isOpenAI and safe_prompt is not None:
```

Comment:
Also since the only two possibilities currently are OpenAI and Mistral, maybe we should use have `isMistral`. Ok with not doing this though.

Reply:
hm I will add it, but it is questionable will we use the same code in the future if we add support for the new AI provider, mistral and openai have quite similar APIs so it was possible to use one flow for both, anthropic has its own separate handler func in the extension